File: python/account_list.py
#!/usr/bin/env python3

# This script assumes an account token.
#
# If all you have is a user token you can run this script, by passing email and password when creating the client
# like so:
# `client = Client(sandbox=True, email=<your_email>, password=<your_password>`

from dnsimple import Client
from auth_token import token

"""
Construct a client instance.

If you want to connect to production omit the sandbox option
"""
client = Client(sandbox=True, access_token=f"{token}")
"""
All calls to client pass through a service. In this case, `client.accounts` is the accounts service

`client.accounts.list_accounts() is the method for retrieving the list of accounts the current authenticated
entity has access to.
"""
response = client.accounts.list_accounts()

account = response.data[0]

print(f'Account= id:{account.id}, email:{account.email}, plan_identifier:{account.plan_identifier}')

File: rust/src/bin/account_list.rs
use dnsimple::dnsimple::new_client;
use dnsimple_rust_sample_code::token_from_file;

fn main() {
    let token = token_from_file();
    account_list(&token);
}

pub fn account_list(token: &str) {
    // Construct a client instance
    //
    // If you want to connect to production set the sandbox argument to false
    let client = new_client(true, token.to_string());

    // All calls to client pass through a service. In this case, `client.accounts()` is the accounts
    // service.
    //
    // `client.accounts().list_accounts` is the method for retrieving the list of accounts for your
    // current credentials via the DNSimple API.
    let response = client.accounts().list_accounts().unwrap().data.unwrap();
    let account = response.first().unwrap();

    println!(
        "Account ID: {}\nAccount Email: {}\nPlan: {}",
        account.id, account.email, account.plan_identifier
    );
}
